src/boot/efi/boot.c

```diff
@@ -2337,6 +2357,46 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
 
         config_load_defaults(&config, root_dir);
 
+        if (setup_mode_enabled() && config.auto_enroll) {
```

Comment:
please split out this whole block into its own function, that exits early if setup mode or auto enroll is off

Reply:
fixed